Mathlib/Algebra/Module/Injective.lean

```diff
@@ -0,0 +1,467 @@
+          (⟨i m, ⟨_, rfl⟩⟩ : LinearMap.range i).2.choose_spec.symm }
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of.inhabited Module.Baer.ExtensionOf.inhabited
+
+/-- Since every nonempty chain has a maximal element, by Zorn's lemma, there is a maximal
+`extension_of i f`. -/
+def extensionOfMax : ExtensionOf i f :=
+  (@zorn_nonempty_partialOrder (ExtensionOf i f) _ ⟨Inhabited.default⟩ fun _ hchain hnonempty =>
+      ⟨ExtensionOf.max hchain hnonempty, ExtensionOf.le_max hchain hnonempty⟩).choose
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max Module.Baer.extensionOfMax
+
+theorem extensionOfMax_is_max :
+    ∀ a : ExtensionOf i f, extensionOfMax i f ≤ a → a = extensionOfMax i f :=
+  (@zorn_nonempty_partialOrder (ExtensionOf i f) _ ⟨Inhabited.default⟩ fun _ hchain hnonempty =>
+      ⟨ExtensionOf.max hchain hnonempty, ExtensionOf.le_max hchain hnonempty⟩).choose_spec
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_is_max Module.Baer.extensionOfMax_is_max
+
+-- Porting note: helper function. Lean looks for an instance of `Sup (Type u)` when the
+-- right hand side is substituted in directly
+@[reducible]
+def supExtensionOfMaxSingleton (y : N) : Submodule R N :=
+  (extensionOfMax i f).domain ⊔ (Submodule.span R {y})
+
+variable {f}
+
+private theorem extensionOfMax_adjoin.aux1 {y : N} (x : supExtensionOfMaxSingleton i f y) :
+    ∃ (a : (extensionOfMax i f).domain) (b : R), x.1 = a.1 + b • y := by
+  have mem1 : x.1 ∈ (_ : Set _) := x.2
+  rw [Submodule.coe_sup] at mem1
+  rcases mem1 with ⟨a, b, a_mem, b_mem : b ∈ (Submodule.span R _ : Submodule R N), eq1⟩
+  rw [Submodule.mem_span_singleton] at b_mem
+  rcases b_mem with ⟨z, eq2⟩
+  exact ⟨⟨a, a_mem⟩, z, by rw [← eq1, ← eq2]⟩
+
+/-- If `x ∈ M ⊔ ⟨y⟩`, then `x = m + r • y`, `fst` pick an arbitrary such `m`.-/
+def ExtensionOfMaxAdjoin.fst {y : N} (x : supExtensionOfMaxSingleton i f y) :
+    (extensionOfMax i f).domain :=
+  (extensionOfMax_adjoin.aux1 i x).choose
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.fst Module.Baer.ExtensionOfMaxAdjoin.fst
+
+/-- If `x ∈ M ⊔ ⟨y⟩`, then `x = m + r • y`, `snd` pick an arbitrary such `r`.-/
+def ExtensionOfMaxAdjoin.snd {y : N} (x : supExtensionOfMaxSingleton i f y) : R :=
+  (extensionOfMax_adjoin.aux1 i x).choose_spec.choose
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.snd Module.Baer.ExtensionOfMaxAdjoin.snd
+
+theorem ExtensionOfMaxAdjoin.eqn {y : N} (x : supExtensionOfMaxSingleton i f y) :
+    ↑x = ↑(ExtensionOfMaxAdjoin.fst i x) + ExtensionOfMaxAdjoin.snd i x • y :=
+  (extensionOfMax_adjoin.aux1 i x).choose_spec.choose_spec
+set_option linter.uppercaseLean3 false in
+#align module.Baer.extension_of_max_adjoin.eqn Module.Baer.ExtensionOfMaxAdjoin.eqn
+
+variable (f)
+
+set_option synthInstance.etaExperiment true in
+-- Porting note: we don't have colon ideals!?
```

Comment:
```lean
/-- `N.colon P` is the ideal of all elements `r : R` such that `r • P ⊆ N`. -/
def colon (N P : Submodule R M) : Ideal R :=
  annihilator (P.map N.mkQ)
#align submodule.colon Submodule.colon
```

In RingTheory.Ideal.Operations (which is ported)

Reply:
So I think this porting note should be updated...